libs/GasBurnerControl/README.md

```diff
@@ -0,0 +1,157 @@
+small delay after each dejam attempt is included in the code.
+
+This state is somewhat a linear state machine on its own realized in the
+private function `_dejam(unsigned int delay_s)`.
+* Wait pre dejam time (0 s after power-up, 60 s after failed ignition,
+or 10 s for conescutive dejam attempts)
+* Press reset button
+* Wait for `DEJAM_DURATION`
+* Release reset button
+* Wait for `POST_DEJAM_DELAY`
+
+I refrained from naming these substates.
+
+If this dejam attempt was not successful as well as if the lock/jam occured
+after ignition was running for some time, the next dejam attempt occurs only
+after "approx. 60 s" in line with the datasheet information. If unsuccessful,
+consecutive dejam attempts will be carried out every "approx. 10 s" until
+it either works or a predefined number of unsuccessful attempts is reached.
+
+Simmilarly, ignition attempts (when ignition started but did not continue to
+running mode --> GBC detects no burning flame), also have a predefined
+max. counter.
+
+
+### `GBC_ERROR`
+
+This state is reached when either a predefined number of consecutive dejam
+attempts failed, or when a consecutive number of ignition attempts failed.
+Whatever occurs first. In this state the GBC is powered down but the
+state machine remains in this state until other user interaction was conducted.
+
+
+## Usage
+
+Call constructor at the beginning of the main code (before `setup` or `loop`) `
+GFAStateMachine gbc(3, 4, 5, 6, 7);`
+
+To start the GBC call `gbc.start();`
+
+As the library is inteded to be interrupt free and non-blocking, call
+`gbc.update()` as often as possible in `loop`. Any state change and error
+detection will only be performed by calling this function!
+
+Finally, if you want to stop the Bruner/GBC call `gbc.stop()`
+
+Thats it :)
+
+
+If you want to get the current state of the state machine without updating,
+call `gba.getState()`.
+
+
+## Options
+
+```
+#define GBC_N_DEJAM_ATTEMPTS 3      // number of unsuccessful dejam attempts before aborting permanently (1-255)
+#define GBC_N_IGNITION_ATTEMPTS 3   // number of unsuccessful ignition attempts before aborting permanently (1-255)
+
+#define GBC_HIGH 0                  // logical high (0: GND, 1: VCC)
+#define GBC_LOW 1                   // logical low (0: GND, 1: VCC)
+
+#define GBC_SERIAL_DEBUG            // if uncommented every update will also send a message over serial (uses lots of RAM)
+
+
+/* TIMER SETTINGS */
+#define GBC_START_DELAY 2           // s, initial delay after powering the GBC
+#define GBC_DEJAM_DELAY 65          // s, wait time until dejamming is possible
+#define GBC_DEJAM_DELAY2 10         // s, wait time between additional dejam attempts
+#define GBC_IGNITION_DELAY 22       // s, time after which ignition should be complete
+#define GBC_DEJAM_DURATION 1000     // ms, duration of button press during dejamming
+#define GBC_POST_DEJAM_DELAY 1000   // ms, delay after dejam button release
```

Comment:
For type safety reasons we should not encourage using macros but make these options just constructor arguments, I doubt adding a few more member variables is going to cost too many bytes.

Reply:
Changed to `struct` with default values in 87de580.
Possibility of logic level inversion changed from macro to struct in 8459162.